tests/tablespaces/Makefile

```diff
@@ -0,0 +1,37 @@
+PGVERSION ?= 10
+
+build:
+	$(MAKE) -C ../../ build-test-pg$(PGVERSION)
+	docker compose build --build-arg "PGVERSION=$(PGVERSION)"
+
+teardown:
+	docker compose down --volumes --remove-orphans
+
+tail:
+	docker compose logs -f
+
+state:
+	docker compose exec monitor pg_autoctl show state
+
+watch:
+	docker compose exec monitor pg_autoctl watch
+
+test: build
+	docker compose up -d node1
+	TEST_NUM="01" docker compose up --exit-code-from=test test
+	docker compose up -d node2
+	TEST_NUM="02" docker compose up --exit-code-from=test test
+	# Pollute the data dir so pg_rewind fails
+	docker compose exec node2 mkdir '/var/lib/postgres/data/base/1/99999999'
+	docker compose exec monitor pg_autoctl perform failover
+	TEST_NUM="03" docker compose up --exit-code-from=test test
+	docker compose pause node1
+	TEST_NUM="04" docker compose up --exit-code-from=test test
+	docker compose unpause node1
+	TEST_NUM="05" docker compose up --exit-code-from=test test
+	docker compose exec monitor pg_autoctl perform failover
+	TEST_NUM="06" docker compose up --exit-code-from=test test
```

Comment:
Let's refactor those bits and have each test with its own Makefile target, and then the main test just depend on them.

``` Makefile
test: build test_01 test_02 ... ;

test_01:
	docker compose up -d node1
  	TEST_NUM="01" docker compose up --exit-code-from=test test

... 
```

Reply:
Sure, the only caveat being that the `failover` one is run twice and make dependencies will miss that, so those will have to not be make targets. (which is fine)